src/main/java/com/estimator/controller/AuthController.java

```diff
@@ -50,10 +50,10 @@ public String getRegisterPage() {
     public String registerUser(@RequestParam String username, @RequestParam String email, @RequestParam String password, RedirectAttributes redirectAttributes) {
         if (userService.findByEmail(email) != null) {
             redirectAttributes.addFlashAttribute("error", "Email already exists");
-            return "redirect:/register";
+            return "redirect:/login";
         }
 
-        userService.registerUser(username,email,passwordEncoder.encode(password),"");
+        userService.registerUser(username,email,passwordEncoder.encode(password), UUID.randomUUID().toString());
         redirectAttributes.addFlashAttribute("message", "Registration successful");
         return "redirect:/login";
```

Comment:
In oauth2Login(), you should check that the /login page correctly handles authorization and redirection errors after a successful login.